libuavcan/src/marshal/uc_bit_array_copy.cpp

```diff
@@ -1,121 +1,62 @@
-        {
-            if (src_offset_modulo > 0U)
-            {
-                unsigned char c = reverse_mask_xor[dst_offset_modulo] & *src++;
-                PREPARE_FIRST_COPY();
-                *dst++ |= c;
-            }
-
-            const unsigned byte_len = src_len / CHAR_BIT;
-            const unsigned src_len_modulo = src_len % CHAR_BIT;
-
-            if (byte_len > 0U)
-            {
-                (void)std::memcpy(dst, src, byte_len);
-                src += byte_len;
-                dst += byte_len;
-            }
-            if (src_len_modulo > 0U)
-            {
-                *dst &= reverse_mask_xor[src_len_modulo];
-                *dst |= reverse_mask[src_len_modulo] & *src;
-            }
-        }
-        else
-        {
-            unsigned bit_diff_ls = 0U;
-            unsigned bit_diff_rs = 0U;
-            unsigned char c = 0U;
-            /*
-             * Begin: Line things up on destination.
-             */
-            if (src_offset_modulo > dst_offset_modulo)
-            {
-                bit_diff_ls = src_offset_modulo - dst_offset_modulo;
-                bit_diff_rs = CHAR_BIT - bit_diff_ls;
-
-                c = static_cast<unsigned char>(*src++ << bit_diff_ls);
-                c = static_cast<unsigned char>(c | (*src >> bit_diff_rs));
-                c &= reverse_mask_xor[dst_offset_modulo];
-            }
-            else
-            {
-                bit_diff_rs = dst_offset_modulo - src_offset_modulo;
-                bit_diff_ls = CHAR_BIT - bit_diff_rs;
-
-                c = static_cast<unsigned char>(*src >> bit_diff_rs & reverse_mask_xor[dst_offset_modulo]);
-            }
-            PREPARE_FIRST_COPY();
-            *dst++ |= c;
-
-            /*
-             * Middle: copy with only shifting the source.
-             */
-            int byte_len = int(src_len / CHAR_BIT);
-
-            while (--byte_len >= 0)
-            {
-                c = static_cast<unsigned char>(*src++ << bit_diff_ls);
-                c = static_cast<unsigned char>(c | (*src >> bit_diff_rs));
-                *dst++ = c;
-            }
-
-            /*
-             * End: copy the remaing bits;
-             */
-            unsigned src_len_modulo = src_len % CHAR_BIT;
-            if (src_len_modulo > 0U)
-            {
-                c = static_cast<unsigned char>(*src++ << bit_diff_ls);
-                c = static_cast<unsigned char>(c | (*src >> bit_diff_rs));
-                c &= reverse_mask[src_len_modulo];
-
-                *dst &= reverse_mask_xor[src_len_modulo];
-                *dst |= c;
-            }
-        }
+    /*
+     * Should never be called on a zero-length buffer. The caller will also ensure that the bit
+     * offsets never exceed one byte.
+     */
+
+    UAVCAN_ASSERT(src_len > 0U);
+    UAVCAN_ASSERT(src_offset < 8U && dst_offset < 8U);
+
+    const size_t last_bit = src_offset + src_len;
+    while (last_bit - src_offset) {
```

Comment:
Sorry, I ran uncrustify just prior to pushing and forgot to actually commit the change. Fixed now.